dds/DCPS/transport/framework/DataLink.cpp

```diff
@@ -188,6 +188,7 @@ void
 DataLink::invoke_on_start_callbacks(bool success)
 {
   const DataLink_rch link(success ? this : 0, inc_count());
+  RepoId remote = GUID_UNKNOWN;
```

Comment:
Why did this definition move?  In general, definition closest to when it's needed is preferred.